pkg/storkctl/schedulepolicy.go

```diff
@@ -88,6 +88,12 @@ func newCreateSchedulePolicyCommand(cmdFactory Factory, ioStreams genericcliopti
 			schedulePolicyName := args[0]
 			var policyItem storkv1.SchedulePolicyItem
 			schedulePolicyType = strings.ToLower(schedulePolicyType)
+			dayOfWeek = strings.ToLower(dayOfWeek)
+			//Validate user input for retain
+			if c.Flags().Changed("retain") && retain <= 0 {
```

Comment:
This check verifies that if user explicitly set the value for retain flag, ensure it is a positive integer, since this flag represents the number of backups triggered as part of this schedule we should retain.
If the user doesn't set the flag value we take the default value for that policy Type.